torch/_dynamo/symbolic_convert.py

```diff
@@ -695,34 +724,42 @@ def get_log_starts_line_log_str(self):
         log_str += f"    {line}"
         return log_str
 
-    def log_starts_line(self):
-        trace_source_log.debug("%s", LazyString(self.get_log_starts_line_log_str))
+    def starts_line(self, lineno):
+        if self.lineno == lineno:
+            return
+        self.lineno = lineno
+        TracingContext.set_current_loc(
+            self.f_code.co_filename, lineno, self.f_code.co_name
+        )
+        if trace_source_log.isEnabledFor(logging.DEBUG):
+            trace_source_log.debug("%s", LazyString(self.get_log_starts_line_log_str))
```

Comment:
I wonder if in python there's a way to prevent argument evaluation when logs are not turned on. In C, you'd do this by macro + shortcircuit like `#DEFINE log LOG_ENABLED && log.xyz()`

Reply:
not really, we could precompute `trace_source_log.isEnabledFor(logging.DEBUG)` to avoid a function call.